### DISCUSSION TITLE
HC_Crypto/Encode/

### DISCUSSION THREAD

**Discussion by giscus** [ProbiusOfficial/Hello-CTF#94] [Show and tell]
# HC_Crypto/Encode/

开源免费、新手友好的CTF(Capture The Flag,夺旗赛)入门教程

https://hello-ctf.com/HC_Crypto/Encode/

**Comment by Xia0-0** [comment]
用到了base64隐写，来着跳着看的，base64那编码过程没看懂，（字节组´･･)ﾉ(._.`)ಠ_ಠ）解码过程懂了，在这挖个坑，之后我再来填

**Reply by Xia0-0** [reply]
解码过程则是编码的逆过程：
把 Base64 字符串去掉等号，转为二进制数
从左到右，8 个位一组，多余位的扔掉，转为对应的 ASCII 码

密文： R28=
去掉=：R28
对应十进制（ASCII 码）：    17     54     60 ←←←←←←
还原位6位二进制 ：       010001 110110 111100
重新分组为8位二进制：     01000111 01101111 00
8位二进制转换为对应ASCII码：  G        o     /(丢弃不完整部分)
得到原文：Go

书中这里好像错了，箭头那行，ASCII码应改为base64编码表，（我当时在这很懵ಠ_ಠ( ´･･)ﾉ(._.`)）

**Comment by Xia0-0** [comment]
解码过程则是编码的逆过程：
把 Base64 字符串去掉等号，转为二进制数
从左到右，8 个位一组，多余位的扔掉，转为对应的 ASCII 码

密文： R28=
去掉=：R28
对应十进制（ASCII 码）：    17     54     60 ←←←←←←
还原位6位二进制 ：       010001 110110 111100
重新分组为8位二进制：     01000111 01101111 00
8位二进制转换为对应ASCII码：  G        o     /(丢弃不完整部分)
得到原文：Go

书中这里好像错了，箭头那行，ASCII码应改为base64编码表，（我当时在这很懵ಠ_ಠ( ´･･)ﾉ(._.`)）

**Reply by Xia0-0** [reply]
评论不能删除么？，重发了一个😂😢ಠ_ಠಠ_ಠಠ_ಠ

**Reply by Randark-JMT** [reply]
> 评论不能删除么？，重发了一个😂😢ಠ_ಠಠ_ಠಠ_ಠ

Github Discussion里面自行删除即可

**Reply by Xia0-0** [reply]
行，欧克，GitHub还不怎么会用，谢谢☆*: .｡. o(≧▽≦)o .｡.:*☆

**Comment by Xia0-0** [comment]
我总结了下，师傅们，帮忙看看对不对☆*: .｡. o(≧▽≦)o .｡.:*☆(❁´◡`❁)

base64：

编码表

ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/
0-------------------------------------------------------------------------------------63


编码→
1.根据ASCII将其转化为8位二进制
2.再转化为6位二进制，不足0来凑
3.转化为十进制
4.映射base64编码表（也就是挨个对应），补齐
finish

解码←
1.把 base64 字符串去掉等号，查base64编码表，变为十进制数（也就是看对应第几个）
2.转为6位二进制数
3.再转化为8位二进制，多余位的扔掉，
4.转为对应的 ASCII 码
finish
